### DISCUSSION TITLE
有没有基于 React 的前端？

### DISCUSSION THREAD

**Discussion by Wybxc** [walinejs/waline#451] [Q&A] [answered]
现在的前端是用 vue 写的，有没有哪位大佬来写一个 React 前端？

**Comment by Mister-Hope** [comment]
V0 是基于react的，体积超出vue很多

**Reply by Wybxc** [reply]
在哪里找？可以作为 React 组件使用吗？

**Comment by lizheming** [comment]
https://github.com/walinejs/waline/tree/ec5a5f8dbc2ff02a28eaa5654980b9bd2e9d49b8/packages/client 这是最后一版的 @waline/client React 版本，对应 https://www.npmjs.com/package/@waline/client/v/0.17.0 。你可以安装使用下，时间比较久了，不确定能否作为组件使用了，可能需要修改下打包相关的逻辑。

**Comment by Wybxc** [comment]
试图用 DOM 方式包装成 React 组件失败了……

可能是因为和 Docusaurus II 冲突了，我再去找找其他的方法

**Reply by Pokoai** [reply]
请问如何将 waline 集合到 Docusaurus2 中呢？

**Reply by Shake-Jarjane** [reply]
> 请问如何将 waline 集合到 Docusaurus2 中呢？

请问你解决了嘛，我也想在docusaurus上配置waline

**Reply by arcosx** [reply]
```js
import React, { PureComponent } from 'react';
import {
  init,
} from '@waline/client';

import '@waline/client/dist/waline.css';

export default class Comment extends PureComponent {
  constructor(props) {
    super(props)
    this._commentRef = React.createRef()
  }

  async componentDidMount() {
    if (typeof window === "undefined") {
      return
    }
    if (!this._commentRef.current) {
      return
    }

    this.Waline = init({
      el: this._commentRef.current,
      serverURL: 'your waline server address',
      visitor: true,
      path: this.props.id,
    })
  }

  render() {
    return <div className="comment-area" ref={this._commentRef} />
  }
}
```

**Reply by Michelle951** [reply]
This helped me a lot. Reeeeeeally thanks!

**Comment by Mister-Hope** [comment, marked as answer]
你可以自行将waline包装成react或者vue组件，waline会直接将自己挂载在你指定的选择器上，所以你自己用组件去封装他是没有难度的。如果你非要把它做成组件的话你可能需要魔改。因为waline本身就已经把自己导出为一个可供调用的函数，而不是一个组件。
